### DISCUSSION TITLE
[Policy] Allow map variables for policy key

### DISCUSSION THREAD

**Discussion by lucaspouzac** [nginx/kubernetes-ingress#5116] [Ideas]
**Is your feature request related to a problem? Please describe.**
Actually, it's not possible to use map on key of policy, if we used nginx CRDs. It's ok if we declare manually this on http-snippets, we cannot re-use it on VirtualServerRoute easily.

**Describe the solution you'd like**
We would like to be able to use a map as a key variable like ${map_(mapName)}

**Describe alternatives you've considered**
Manual declaration, without CRDs usage.
```
geo $ip_whitelist {
    default             limited;
    8.8.8.8/32         nolimit;  
    127.0.0.1/32     nolimit; 
}

map $ip_whitelist $limit {
    limited  $binary_remote_addr;
    nolimit  "";
}

limit_req_zone $limit zone=reqWhitelist:10m rate=10r/s;
```

**Additional context**
https://github.com/nginxinc/kubernetes-ingress/blob/v3.4.2/pkg/apis/configuration/validation/policy.go#L479

**Comment by vepatel** [comment]
hi @lucaspouzac we're looking into this and will update it in due course. We want to make sure whatever the solution is, it should work for both Ingress and VirtualServer.

**Comment by brianehlert** [comment]
Accepting a "map block" as Policy is essentially a sting blob.  It is no better than or different than a snippet.
But rather than being in-line in context, it becomes a reusable object that has context.

I think the way to think about this is more like a programmability capability such as 'action' which is a capability of the VirtualServer custom resource.
